File: src/promo-prompts.js
/**
 * 各平台推广文案写作规范（供 AI system prompt 使用）
 * 参考：Pipepost Show HN Launch Kit、developer-marketing-guides、
 *       readme-SVG/repo-promotion-guide、小红书爆款四段结构、知乎高赞回答体
 */

export const PROMO_JSON_SCHEMA = [
  '{',
  '  "positioning": "一句话定位（结果导向，不是功能列表）",',
  '  "targetUsers": ["目标用户1", "目标用户2"],',
  '  "strongestAngles": ["最适合推广的切入点1", "切入点2"],',
  '  "promotionStrategy": {',
  '    "coreAngle": "基于来源证据的主推广角度",',
  '    "contentGraph": [{"node": "problem|method|evidence|visual|caveat|action", "claim": "内容节点", "source": "README/PDF/figure/table/demo/command"}],',
  '    "audienceSegments": [{"audience": "目标读者", "platform": "适合平台", "reason": "为什么"}],',
  '    "platformAdaptation": [{"platform": "xhs|zhihu|wechat|showHn|twitter|productHunt", "format": "格式", "tone": "语气", "visual": "配图/截图建议", "avoid": ["不要写什么"]}],',
  '    "visualNarrative": [{"asset": "来源截图/论文图/生成图", "supportsClaim": "支撑的观点"}],',
  '    "qualityRubric": {',
  '      "fidelity": {"checks": ["事实准确性/核心贡献/术语一致性检查"], "risks": ["可能失真的点"], "improvements": ["发布前如何补证据"]},',
  '      "engagement": {"checks": ["开头钩子/叙事清晰/CTA/受众吸引力检查"], "risks": ["可能太泛或太像模板的点"], "improvements": ["如何让读者更愿意点开"]},',
  '      "alignment": {"checks": ["平台语气/节奏/标签/图文配合检查"], "risks": ["平台不适配的点"], "improvements": ["如何按平台改写或换图"]}',
  '    },',
  '    "reviewGate": {"fidelityQuestions": ["事实核查问题"], "engagementQuestions": ["吸引力检查"], "platformQuestions": ["平台适配检查"]}',
  '  },',
  '  "promotions": {',
  '    "xiaohongshu": {',
  '      "titles": ["标题1≤20字", "标题2", "标题3"],',
  '      "markdown": "完整 Markdown 文件正文（可直接保存为 promo-xhs.md）",',
  '      "tags": ["#标签1", "#标签2"],',
  '      "publishNotes": "发布建议（配图、发布时间等）"',
  '    },',
  '    "wechatMoments": {',
  '      "markdown": "完整 Markdown（含2-3种朋友圈风格，每种可直接复制）"',
  '    },',
  '    "zhihu": {',
  '      "suggestedQuestions": ["适合回答的知乎问题1", "问题2"],',
  '      "markdown": "完整 Markdown 回答/专栏正文（1000-2000字）"',
  '    },',
  '    "showHn": {',
  '      "title": "Show HN: Name – plain English description",',
  '      "markdown": "完整 Markdown（含标题、URL、首条评论模板、回复话术）"',
  '    },',
  '    "twitter": {',
  '      "markdown": "完整 Markdown（3-5条 thread，英文）"',
  '    },',
  '    "reddit": {',
  '      "markdown": "完整 Markdown（Reddit/V2EX 标题+正文）"',
  '    },',
  '    "productHunt": {',
  '      "markdown": "完整 Markdown（tagline + maker first comment）"',
  '    }',
  '  },',
  '  "launchSequence": [{"order": 1, "channel": "渠道", "reason": "为什么先在这个渠道发", "ready": true}]',
  '}'
].join('\n');

export function buildPromoSystemPrompt() {
  return [
    '你是项目/论文发布内容策划 + 多平台内容主编，专门把开源仓库、论文 PDF、README 和来源证据转成可人工审核的推广 Markdown 文件。',
    '你会收到 README 原文、安装命令、Demo 证据、论文/PDF 摘要、规则扫描结果、launchRisks、topFixes 等。请独立阅读 evidence，不要机械复述体检分数或指标数字。',
    '',
    '## 可信度铁律（最高优先级）',
    '- 禁止编造：star 增长数量/时间线（如「第二天多了 N 个 star」）、用户数量、媒体报道、他人评价',
    '- 禁止编造：未在 evidence 出现的功能、平台、集成、案例',
    '- 允许使用：README/evidence 中可核实的事实；「我跑了一下」「输出里提到」等可验证表述',
    '- 若 launchRisks / topFixes 指出短板（占位符、缺 GIF、README 问题），中文推广应诚实提及或回避过度承诺，英文 Show HN 首评必须承认 1-2 个局限',
    '- 推广 Source2Launch 自身时：像维护者在记录工具进展，不是营销号吹产品；少说「神器/必备/yyds」',
    '',
    '## 核心原则',
    '- 写「用户场景」而非「功能清单」：读者读完要知道「我能用它做什么、30 秒怎么试」',
    '- 一个平台只聚焦 1 个最打动人的使用场景，不要罗列 8 维评分',
    '- 中文平台用中文，英文平台用英文；语气像真人分享，不像 AI 模板',
    '- 每个平台的 markdown 字段必须是完整、可独立保存的 Markdown 文件（含标题层级、段落、列表、代码块）',
    '- Markdown 段落之间必须有空行；## 标题前后必须空行',
    '',
    '## 反 AI 味（必须遵守）',
    '禁用：首先/其次/最后、综上所述、值得一提的是、在当今数字化、赋能、助力、',
    '       浅谈、分享一个、推荐一个、你觉得呢、欢迎留言、一键神器、必备工具',
    '多用：我、亲测、说实话、踩坑、后悔没早知道、一行命令、打开 README 前 10 秒',
    '',
    '## 受欢迎内容的共性',
    '- 小红书：像「记录一次排查过程」，不是「安利产品」；结尾留开放式问题或「你们仓库首屏怎么写？」',
    '- 微信：像发给同事的一句话，可略口语、可省略主语，不要三段论',
    '- 知乎：70% 讲方法论/踩坑，30% 提工具；产品名全文不超过 5 次；文末 #话题',
    '- Show HN：Problem 要具体（45 分钟手动改 README）；主动写 2 个 limitations',
    '',
    '## AutoPR-style 三轴审核',
    '- Fidelity：检查事实准确性、核心贡献覆盖、术语/作者/标题/命令是否与来源一致；宁可少写，不要补编。',
    '- Engagement：检查开头是否具体、叙事是否有逻辑、CTA 是否指向读论文/试命令/看仓库；不要用空泛爆款钩子。',
    '- Alignment：检查平台语气、节奏、标签、图片比例和图文配合是否匹配；不要把同一段文字复制到所有平台。',
    '- 每次输出都要填写 promotionStrategy.qualityRubric，让人工审核能看到风险和改进动作。',
    '',
    '---',
    '',
    buildXhsGuide(),
    '',
    '---',
    '',
    buildWechatGuide(),
    '',
    '---',
    '',
    buildZhihuGuide(),
    '',
    '---',
    '',
    buildShowHnGuide(),
    '',
    '---',
    '',
    buildTwitterGuide(),
    '',
    '---',
    '',
    buildRedditGuide(),
    '',
    '---',
    '',
    '只输出严格 JSON，不要 Markdown 代码块，不要解释 JSON 之外的内容。'
  ].join('\n');
}

export function buildPromoUserPrompt(payload, options = {}) {
  const platform = options.platform ?? 'all';
  const platformHint = platform === 'all'
    ? '请生成所有平台的完整 markdown 字段。launchSequence 按仓库准备度给出发布顺序。'
    : `重点打磨 ${platform} 对应平台，其他平台给出简短可用 markdown。`;
  const evidenceBrief = buildEvidenceBrief(payload);
  const briefSection = options.briefSection ?? '';

  return [
    '请基于以下来源数据，生成各平台可直接发布的推广 Markdown 文件内容。',
    '',
    '## 写作时必须引用的真实证据（不可违背）',
    evidenceBrief,
    '',
    briefSection,
    briefSection ? '' : null,
    '写作时请参考优秀开源项目的 Launch Kit 写法：',
    '- 规划层：先抽取来源材料，再合成主推广角度，最后按平台改写，并给出 fidelity / engagement / platform alignment 三轴审核',
    '- 三轴审核必须写入 promotionStrategy.qualityRubric，不要只给笼统建议',
    '- Show HN：个人故事 + 具体问题 + 技术切入点 + 首条评论 + 2 条 limitations',
    '- 小红书：像记录一次排查 README 的过程，不要写虚假 star 增长',
    '- 微信：像发给同事，短、口语、带一条命令',
    '- 知乎：先讲「开源项目/论文如何讲清楚」的方法论，工具作为附录提及',
    '',
    'JSON 输出结构必须是：',
    PROMO_JSON_SCHEMA,
    '',
    platformHint,
    '',
    'markdown 字段要求：',
    '- 以 # 一级标题开头（如 `# 项目名 · 小红书推广`）',
    '- 包含 `> 元信息` 引用块说明用途',
    '- 正文可直接复制到对应平台发布',
    '- 安装命令用 ```sh 代码块，第一行必须与 project.installCommand 完全一致',
    '- 第二行可写 `source2launch optimize . --output launch-assets/`（不要使用旧命令或不存在的别名）',
    '- 段落之间空一行；禁止「以下是文案」「数据显示」等 meta/编造统计',
    '- 知乎必须含 ## 快速上手、## 建议回答的问题（或 suggestedQuestions 字段）、文末 #话题',
    '- 小红书 markdown 必须含 ## 标题备选（3条）和 2-3 个 emoji',
    '- 微信每种风格正文用空行分段（2-4 行），链接单独一行放末尾',
    '',
    '完整仓库数据：',
    JSON.stringify(payload, null, 2)
  ].filter(Boolean).join('\n');
}

export function buildEvidenceBrief(payload) {
  const project = payload.project ?? {};
  const lines = [];
  lines.push(`- 项目：${project.name ?? 'unknown'}`);
  if (project.description) lines.push(`- 描述：${project.description}`);
  if (project.installCommand) lines.push(`- 安装命令（必须原样使用）：\`${project.installCommand}\``);
  if (project.repositoryUrl) lines.push(`- 仓库：${project.repositoryUrl}`);
  if (project.topics?.length) lines.push(`- Topics：${project.topics.join(', ')}`);

  if (payload.heuristicScore) {
    lines.push('- 本地资料检查：已完成，仅作 CI / 资料完整度参考；推广正文不要展示分数或等级');
  }

  const opening = payload.evidence?.readmeOpening;
  if (opening) {
    const snippet = String(opening).replace(/\s+/g, ' ').slice(0, 200);
    lines.push(`- README 开头片段：${snippet}…`);
  }

  const risks = payload.evidence?.launchRisks;
  if (Array.isArray(risks) && risks.length > 0) {
    lines.push('- launchRisks（可诚实提及或用于 Show HN limitations）：');
    for (const risk of risks.slice(0, 3)) lines.push(`  - ${risk}`);
  }

  if (Array.isArray(payload.topFixes) && payload.topFixes.length > 0) {
    lines.push('- 优先改进项（可用于「我还在完善…」）：');
    for (const fix of payload.topFixes.slice(0, 3)) {
      lines.push(`  - ${fix.fix ?? fix.message ?? fix}`);
    }
  }

  const strongChecks = (payload.checks ?? [])
    .filter((check) => check.score / check.max >= 0.75)
    .slice(0, 3);
  if (strongChecks.length > 0) {
    lines.push('- 可引用的真实亮点（转成场景句，不要照搬 summary）：');
    for (const check of strongChecks) {
      lines.push(`  - ${check.label}：${check.summary}`);
    }
  }

  return lines.join('\n');
}

function buildXhsGuide() {
  return [
    '## 小红书（xiaohongshu.markdown）',
    '',
    '结构参考爆款四段式：',
    '1. **钩子**（前2行）：痛点或反常识，如「做了3个月开源，star 一直个位数？」',
    '2. **共鸣**：具体场景，「你是不是也…」',
    '3. **价值**（60%篇幅）：聚焦 1 个核心用法 + 真实细节（命令、路径、效果）',
    '4. **收尾**：留开放式问题或「你们怎么写 README 首屏？」，不要硬 CTA',
    '',
    '标题 titles：≤20字，含数字/身份/反差，禁用「分享」「推荐」「浅谈」',
    '正文：300-600字，短段落（1-3句），第一人称，可 2-3 个 emoji',
    '禁止：虚假 star 增长、编造用户反馈、堆砌 8 维评分数字',
    'tags：5-8个，混合大流量（#开源 #程序员）和精准标签',
    '',
    'markdown 文件结构示例：',
    '# 项目名 · 小红书推广',
    '> 可直接复制发布',
    '## 标题备选',
    '- 标题1',
    '## 正文',
    '（四段式正文）',
    '## 标签',
    '#开源 #GitHub …',
    '## 发布提示',
    '（配图建议、最佳发布时间）'
  ].join('\n');
}

function buildWechatGuide() {
  return [
    '## 微信朋友圈（wechatMoments.markdown）',
    '',
    '参考 ShareCraft / 朋友圈安利写法，提供 2-3 种风格，每种可直接复制：',
    '',
    '**风格 A · 朋友安利**：我发现… + 为什么有用 + 链接',
    '**风格 B · 开源心得**：如果你也在做开源… + 可借鉴的点',
    '**风格 C · 短帖**：一句话 + 命令 + 链接（80字内）',
    '',
    '要求：第一人称，说感受不说指标，不要「今天推荐一个开源项目」式公文',
    '每条 60-150 字，链接单独一行放末尾；句子之间用空行分隔（像真发朋友圈）',
    '禁止：体检分数、8维评分、star 增长承诺、三连请求',
  ].join('\n');
}

function buildZhihuGuide() {
  return [
    '## 知乎（zhihu.markdown）',
    '',
    '参考知乎高赞回答四段式：',
    '1. **开篇**（100-200字）：反常识/痛点/个人经历引入',
    '2. **核心**（2-4个论点）：每点有观点 + 证据 + 可操作步骤',
    '3. **实操**：安装命令、典型用法、README 亮点',
    '4. **升华**：总结 + 邀请讨论',
    '',
    '自然嵌入 2-3 个词：指南、教程、案例分析、经验分享',
    'suggestedQuestions：3 个适合回答的真实知乎问题',
    '正文 1000-1800 字；产品名全文不超过 5 次；70% 讲方法论，30% 讲工具',
    '必须包含 ## 快速上手 和文末 #开源 #GitHub 等话题行',
  ].join('\n');
}

function buildShowHnGuide() {
  return [
    '## Show HN（showHn.markdown）',
    '',
    '参考 Pipepost / developer-marketing-guides / repo-promotion-guide：',
    '',
    '标题：`Show HN: [Name] – [plain English what it does]`（≤80字符，诚实描述，无 clickbait）',
    '',
    'markdown 包含：',
    '## 提交信息',
    '- Title / URL / 提交时间建议（Tue-Thu 8-10am ET）',
    '## 首条评论（OP Comment）',
    '个人口吻：我遇到什么问题 → 现有方案为何不够 → 我做了什么 → 技术亮点 → 邀请反馈',
    '参考 Pipepost 首评：Problem → Approach → Differentiation → Try it → Discussion',
    '## 常见问题回复模板',
    '3-5 条 anticipated objections 的回复草稿',
    '',
    '风格：第一人称 I/we，承认局限，不请求 upvote，技术细节具体'
  ].join('\n');
}

function buildTwitterGuide() {
  return [
    '## Twitter/X（twitter.markdown）',
    '',
    '3-5 条 thread，英文：',
    '1/ Hook：问题或结果',
    '2/ What it is + who it is for',
    '3/ Demo command or screenshot description',
    '4/ Link to repo',
    '5/ Ask for feedback / #buildinpublic',
    '',
    '每条 ≤280 字符，具体不空洞'
  ].join('\n');
}

function buildRedditGuide() {
  return [
    '## Reddit / V2EX（reddit.markdown）',
    '',
    '标题：`[Project] Name — what it does in one line`',
    '正文：问题背景 → 解决方案 → 快速开始 → 链接',
    '语气克制，像社区成员分享而非广告',
    'V2EX 节点建议：分享创造 / 推广'
  ].join('\n');
}

File: promoagent/promo_prompts.py
from __future__ import annotations

import json
from typing import Any


# ---------------------------------------------------------------------------
# Minimal JSON schema — structure only, platforms decided by AI
# ---------------------------------------------------------------------------

PROMO_JSON_SCHEMA = """{
  "positioning": "一句话定位（结果导向，具体可验证）",
  "targetUsers": ["目标用户"],
  "strongestAngles": ["最适合推广的切入点"],
  "promotionStrategy": {
    "coreAngle": "基于来源证据的主推广角度",
    "contentGraph": [{"node": "problem|evidence|action", "claim": "内容", "source": "来源"}],
    "qualityRubric": {
      "fidelity": {"checks": ["..."], "risks": ["..."], "improvements": ["..."]},
      "engagement": {"checks": ["..."], "risks": ["..."], "improvements": ["..."]},
      "alignment": {"checks": ["..."], "risks": ["..."], "improvements": ["..."]}
    }
  },
  "promotions": {
    "<platform_key>": {
      "markdown": "完整可发布的 Markdown 正文",
      "publishNotes": "发布建议"
    }
  },
  "launchSequence": [{"order": 1, "channel": "渠道", "reason": "原因"}]
}"""

# platform_key 示例：xiaohongshu / zhihu / wechatMoments / wechatArticle /
# twitter / showHn / productHunt / linkedin / reddit / weibo
# 只生成适合这份内容的平台，不需要覆盖全部


# ---------------------------------------------------------------------------
# Preset expansion — no hardcoded content, treats preset names as intent hints
# ---------------------------------------------------------------------------

def expand_presets(names: list[str]) -> str:
    """Convert preset names into intent hints for the AI.

    No hardcoded instructions — just passes the names as context so the AI
    can interpret them based on its own understanding.
    """
    if not names:
        return ""
    hint = "、".join(n.strip() for n in names if n.strip())
    return (
        f"写作风格偏好（请根据自己的理解应用）：{hint}\n"
        "以上是用户希望的写作风格参考，请在保证内容基于来源证据的前提下，适当融入这些风格。"
    )


# ---------------------------------------------------------------------------
# System prompt — principles only, no hardcoded platform rules
# ---------------------------------------------------------------------------

def build_promo_system_prompt() -> str:
    return "\n".join([
        "你是推广内容主编，把任何来源（开源项目、论文、餐厅、产品、活动、服务…）转成各平台推广文案。",
        "你已经熟悉各平台的内容风格、格式要求和用户习惯，请自主运用这些知识。",
        "",
        "## 好内容 vs 平庸内容",
        "平庸内容：描述产品特征，告诉读者「这个东西很好」。",
        "好内容：让目标读者在特定场景下产生「这说的就是我」的感觉，知道下一步该做什么。",
        "",
        "## 写作前必须想清楚",
        "- 目标读者是谁？他们在什么场景下看到这条内容？看完之后你希望他们做什么？",
        "- 来源证据里最有力的一个事实是什么？（价格、效果、独特性、反差感…）",
        "- 第一句话如何让目标读者停下来？（具体场景 > 数字反差 > 情绪共鸣 > 悬念问题）",
        "",
        "## 核心原则",
        "- 只使用来源证据中可核实的事实。无法验证的内容写 caveat，不要补编。",
        "- 根据内容特点和受众，自主判断哪些平台最适合，选择 2-5 个平台生成内容。",
        "- 每个平台的内容要有明显差异：不只是改改措辞，结构和切入角度都要不同。",
        "- 三轴审核（fidelity / engagement / alignment）必须填写。",
        "",
        "只输出严格 JSON，不要 Markdown 代码块，不要 JSON 之外的内容。",
    ])


# ---------------------------------------------------------------------------
# User prompt
# ---------------------------------------------------------------------------

def build_promo_user_prompt(
    payload: dict[str, Any],
    *,
    platform: str = "all",
    brief_section: str = "",
    examples: list[str] | None = None,
) -> str:
    platform_hint = (
        "请为这份内容选择最合适的 2-5 个平台，生成完整的 markdown 内容。"
        if platform == "all"
        else f"重点生成 {platform} 的内容，同时为 1-2 个其他合适的平台生成简短版本。"
    )

    brief = build_evidence_brief(payload)
    source_type = payload.get("inputType") or payload.get("source") or ""
    is_sparse = source_type == "text" or len(brief) < 200

    parts = [
        "请基于以下来源证据，生成各平台可直接使用的推广内容。",
        "",
        "## 来源证据（必须引用，不可违背）",
        brief,
        "",
    ]

    if is_sparse:
        parts.extend([
            "## 证据稀薄时的处理方式",
            "当前来源信息较少，但这不意味着无法写出好内容。",
            "请从已有信息中挖掘隐含价值：",
            "- 价格/时间/地点信息背后的「读者意义」是什么？",
            "- 产品/服务名称里有没有暗示的受众、场景或差异化？",
            "- 能从描述词语中推断出哪些体验性细节？",
            "对于无法从证据推断的内容，在 reviewGate 里标注「需要补充」，不要编造。",
            "",
        ])

    if brief_section:
        parts.extend([brief_section, ""])

    # Few-shot examples from Stage 1
    if examples:
        example_section = format_examples_for_prompt(examples, platform)
        if example_section:
            parts.extend([example_section, ""])

    parts.extend([
        "## 写作流程（按顺序执行）",
        "",
        "**Step 1 — 挖掘真正的推广角度**",
        "不要直接描述产品特征，而是找到它背后的「读者价值」：",
        "- 它解决了一个什么具体的烦恼？还是满足了一个什么具体的欲望？",
        "- 目标读者在什么场景下最需要这个？",
        "- 来源证据里有没有一个数字、细节或反差，能让人一眼记住？",
        "把找到的最强角度写入 promotionStrategy.coreAngle。",
        "",
        "**Step 2 — 策划内容图谱**",
        "用证据中的具体材料（命令/价格/地址/图片/数据/评价）填充 contentGraph，",
        "标注每个 claim 来自哪里，确保每个平台都有可引用的原始证据。",
        "",
        "**Step 3 — 按平台写，真正有差异**",
        "不同平台的差异不只是语气——核心切入角度也应该不同：",
        "- 同一家餐厅：小红书切「隐藏宝藏发现感」，知乎切「性价比分析」，微信切「这个周末去哪吃」",
        "- 同一个工具：Show HN 切「解决了什么技术痛点」，小红书切「这个命令帮我省了多少时间」",
        "",
        "**Step 4 — 三轴审核**",
        "- Fidelity：每个 claim 都能在来源证据中找到依据？",
        "- Engagement：第一句话是否具体？读者能否在 3 秒内判断「这和我有关」？",
        "- Alignment：语气、节奏、格式是否真的像这个平台的原生内容？",
        "",
        platform_hint,
        "",
        "JSON 输出结构：",
        PROMO_JSON_SCHEMA,
        "",
        "完整来源数据：",
        json.dumps(payload, ensure_ascii=False, indent=2),
    ])
    return "\n".join(parts)


# ---------------------------------------------------------------------------
# Payload builder  (moved here from ai.py — data prep belongs in promo layer)
# ---------------------------------------------------------------------------

def build_promo_payload(result: dict[str, Any]) -> dict[str, Any]:
    """Extract the fields that AI prompts need from a full analyze_target() result."""
    return {
        "project":   result.get("project", {}),
        "evidence":  result.get("evidence", {}),
        "target":    result.get("target"),
        "source":    result.get("source"),
        "inputType": result.get("inputType"),
    }


# ---------------------------------------------------------------------------
# Evidence brief
# ---------------------------------------------------------------------------

def build_evidence_brief(payload: dict[str, Any]) -> str:
    project = payload.get("project", {})
    evidence = payload.get("evidence", {})
    ctx = evidence.get("additionalContext", {})

    lines = [f"- 推广主体：{project.get('name', '（未命名）')}"]

    desc = project.get("description") or ""
    if desc:
        lines.append(f"- 核心描述：{desc[:400]}")

    cta = project.get("cta") or project.get("installCommand")
    if cta:
        lines.append(f"- 行动号召（必须原样使用）：{cta}")

    if project.get("repositoryUrl"):
        lines.append(f"- 代码仓库：{project['repositoryUrl']}")
    if project.get("homepage"):
        lines.append(f"- 主页/网站：{project['homepage']}")

    if project.get("topics"):
        lines.append(f"- 关键词：{', '.join(project['topics'][:8])}")

    if project.get("stars") is not None:
        lines.append(f"- GitHub Stars：{project['stars']}")

    opening = evidence.get("opening") or evidence.get("readmeOpening") or ""
    if opening and opening != desc:
        lines.append(f"- 内容概述：{opening[:300]}")

    headings = evidence.get("headings") or []
    h2 = [h["text"] for h in headings if isinstance(h, dict) and h.get("level") == 2][:8]
    if h2:
        lines.append(f"- 主要章节/功能：{' / '.join(h2)}")

    key_actions = evidence.get("keyActions") or evidence.get("installCommands") or []
    extra = [a for a in key_actions if a != cta][:3]
    if extra:
        lines.append(f"- 使用方式示例：{' | '.join(extra)}")

    proofs = evidence.get("proofPoints") or []
    if proofs:
        lines.append("- 质量证明：")
        for p in proofs[:3]:
            lines.append(f"  - {p}")

    if ctx:
        ctx_items = []
        for k, v in ctx.items():
            if v and k != "interactiveQA":
                ctx_items.append(f"{k}：{v}")
        if ctx_items:
            lines.append(f"- 补充信息：{' / '.join(ctx_items)}")

    # Interactive Q&A from user
    qa = ctx.get("interactiveQA")
    if qa:
        lines.extend(["- 用户补充的信息：", qa])

    clips = evidence.get("documentClips") or []
    if clips:
        lines.append("- 文档摘要：")
        for clip in clips[:2]:
            text = clip.get("text", "") if isinstance(clip, dict) else str(clip)
            lines.append(f"  > {text[:150].replace(chr(10), ' ')}")

    risks = evidence.get("launchRisks") or []
    if risks:
        lines.append("- 注意事项（可用于 limitations）：")
        for risk in risks[:3]:
            msg = risk.get("message") if isinstance(risk, dict) else str(risk)
            lines.append(f"  - {msg}")

    return "\n".join(lines)


# ---------------------------------------------------------------------------
# Stateless refinement messages  (used by mcp_server.s2l_refine)
# ---------------------------------------------------------------------------

def build_refine_messages(
    content: str,
    feedback: str,
    platform: str = "",
) -> list[dict[str, str]]:
    """Build a one-shot message list for stateless content refinement.

    Unlike ``ai.refine_content()`` (which requires saved conversation history),
    this is suitable for stateless callers such as the MCP server tools.
    """
    platform_hint = f"（目标平台：{platform}）" if platform else ""
    return [
        {"role": "system", "content": build_promo_system_prompt()},
        {
            "role": "user",
            "content": (
                f"以下是已生成的推广文案{platform_hint}，请根据反馈进行改进。\n\n"
                f"**原文案：**\n{content}\n\n"
                f"**改进要求：**\n{feedback}\n\n"
                "请直接输出改进后的完整文案，无需解释修改原因。"
            ),
        },
    ]


# ---------------------------------------------------------------------------
# Few-shot example formatter  (moved here from examples.py to break the
# promo_prompts → examples → ai → promo_prompts import cycle)
# ---------------------------------------------------------------------------

def format_examples_for_prompt(examples: list[str], platform: str = "all") -> str:
    """Format Stage-1 reference examples as a few-shot section for the user prompt."""
    if not examples:
        return ""
    platform_label = platform if platform else "通用"
    lines = [
        f"## 参考示例（{platform_label} 优质内容风格参考）",
        "",
        "请学习以下示例的写作风格、结构和语感，但**不要复制内容**。",
        "你的输出必须完全基于来源证据，与示例内容无关。",
        "",
    ]
    for i, example in enumerate(examples, 1):
        lines += [f"### 示例 {i}", "", example.strip(), ""]
    lines += ["---", ""]
    return "\n".join(lines)
